tests/runtime/filter_aws.c

```diff
@@ -582,11 +582,257 @@ void flb_test_aws_ec2_tags_value_500() {
     set_output(NULL);
 }
 
+void flb_test_aws_ec2_tags_include() {
+    int ret;
+    int bytes;
+    char *p = "[0, {\"log\": \"hello, from my ec2 instance\"}]";
+    flb_ctx_t *ctx;
+    int in_ffd;
+    int out_ffd;
+    int filter_ffd;
+    struct flb_lib_out_cb cb_data;
+    struct flb_aws_client_generator *client_generator;
+    struct flb_filter_aws_init_options *ops;
+    flb_sds_t name;
+    struct flb_aws_client_mock_request_chain *request_chain;
+    char *output = NULL;
+    char *result;
+
+    request_chain = FLB_AWS_CLIENT_MOCK(
+        response(
+            expect(URI, "/latest/meta-data/tags/instance"),
+            expect(METHOD, FLB_HTTP_GET),
+            set(STATUS, 200),
+            set(PAYLOAD, "Name\nCUSTOMER_ID"),
+            set(PAYLOAD_SIZE, 16)
+        ),
+        response(
+            expect(URI, "/latest/meta-data/tags/instance/Name"),
+            expect(METHOD, FLB_HTTP_GET),
+            set(STATUS, 200),
+            set(PAYLOAD, "my_ec2_instance"),
+            set(PAYLOAD_SIZE, 15)
+        ),
+        response(
+            expect(URI, "/latest/meta-data/tags/instance/CUSTOMER_ID"),
+            expect(METHOD, FLB_HTTP_GET),
+            set(STATUS, 200),
+            set(PAYLOAD, "70ec5c04-3a6e-11ed-a261-0242ac120002"),
+            set(PAYLOAD_SIZE, 36)
+        )
+    );
+    flb_aws_client_mock_configure_generator(request_chain);
+
+    client_generator = flb_aws_client_get_mock_generator();
+    ops = flb_calloc(1, sizeof(struct flb_filter_aws_init_options));
+    name = flb_sds_create_size(100);
+    name = flb_sds_printf(&name, "flb_test_aws_ec2_tags_include");
+    ops->client_generator = client_generator;
+    ops->name = name;
```

Comment:
Sorry I can't remember, where is hte name actually used? I bet it should be safe to just use a string in the test function like:

```
ops->name = "flb_test_aws_ec2_tags_include";
```

Reply:
https://github.com/fluent/fluent-bit/pull/6866/commits/2af2b454da0dfa61eceed34c15ff7a4b5b9d3a1d